python/iceberg/core/filesystem/util.py

```diff
@@ -33,21 +31,8 @@ def get_fs(path, conf, local_only=False):
         if parsed_path.scheme in ["", "file"]:
             return LocalFileSystem.get_instance()
         elif parsed_path.scheme in ["s3", "s3n"]:
-            if conf.get("hive.aws_iam_role") is not None:
-                key, secret, token = get_sts_session_keys(conf.get("hive.aws_iam_role"))
-                return S3FileSystemWrapper(key=key, secret=secret, token=token)
-            return S3FileSystemWrapper()
+            return S3FileSystemWrapper(conf=conf)
```

Comment:
let me review this, it's been a little bit since I was in this part of the code.  I think you are right that this should be made consistent between the two.